### PR TITLE
`ultralytics 8.2.63` refactor `FastSAMPredictor`

### PR DESCRIPTION
**Pull request by maycuatroi1**
# Changes:
- Refactored the `FastSAMPredictor` code to now directly inherit from `SegmentationPredictor`.
- Added a protected function `_process_full_box` in `SegmentationPredictor` to handle post-processing for `SegmentationPredictor`.

# Issues:
- Code duplications

# Tested script:
```python
from ultralytics import FastSAM
from ultralytics.models.fastsam import FastSAMPrompt

# Define an inference source
source = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/Dog_Breeds.jpg/800px-Dog_Breeds.jpg"

# Create a FastSAM model
model = FastSAM("FastSAM-x.pt")  # or FastSAM-x.pt

# Run inference on an image
everything_results = model(source, device="cpu", retina_masks=True, imgsz=1024, conf=0.2, iou=0.2)

# Prepare a Prompt Process object
prompt_process = FastSAMPrompt(source, everything_results, device="cpu")
ann = prompt_process.text_prompt(text="a photo of a dog")
prompt_process.plot(annotations=ann, output="./")
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR enhances the segmentation prediction process in Ultralytics YOLO by refining how full bounding boxes are processed for the FastSAMPredictor.

### 📊 Key Changes
- Switch `FastSAMPredictor` from extending `DetectionPredictor` to `SegmentationPredictor`.
- Simplified post-processing logic by integrating a new `_process_full_box` method to handle prediction updates.
- Refined bounding box adjustments and non-max suppression integration.

### 🎯 Purpose & Impact
- **Enhanced Segmentation Accuracy**: By focusing on segmentation-specific predictions, the new class structure improves the handling of segmentation tasks.
- **Streamlined Code**: Simplifies the prediction pipeline, making the code easier to maintain and extend.
- **Improved Performance**: More efficient handling of predictions can result in faster processing times and more accurate results for segmentation models.

😊 Users can expect more robust and accurate segmentation predictions with optimized performance!

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR introduces changes to improve issue and pull request automation, refactors prediction classes, and updates documentation for better clarity.

### 📊 Key Changes
- **Workflow Updates**: 
  - Added triggers for issue events (`opened`, `edited`).
  - Enabled auto-labeling for issues and pull requests.
- **Documentation**:
  - Refined content in `docs/en/reference/models/fastsam/utils.md`.
- **Code Refactoring**:
  - Changed the base class for `FastSAMPredictor` from `DetectionPredictor` to `SegmentationPredictor`.
  - Removed deprecated methods and consolidated post-processing logic in `ultralytics/models/fastsam/predict.py`.
- **Version Update**:
  - Bumped version from `8.2.62` to `8.2.63`.

### 🎯 Purpose & Impact
- **Improved Automation**:
  - 🛠️ **Efficiency**: Automating issue labeling and pull request handling saves time and ensures consistency.
- **Enhanced Documentation**:
  - 📚 **Clarity**: Simplified and clarified documentation makes it easier for users to understand and use the functions.
- **Streamlined Predictions**:
  - 🚀 **Performance**: Refactoring predictors improves performance and maintains compatibility with segmentation tasks, ensuring users get faster and more accurate results.
- **Versioning**:
  - 📈 **Updates**: The version bump helps users quickly identify that new changes or fixes are included in this release.

### CODE REVIEW

**Comment by maycuatroi** [comment]
I have read the CLA Document and I sign the CLA

**Comment by glenn-jocher** [comment]
@maycuatroi awesome, thank you!

@Laughing-q can you review this please? Thanks!

**Comment by Laughing-q** [comment]
@maycuatroi Hi! Thanks for the PR! It helps reduce the duplication of the code, however I prefer not to introduce a new function i.e `_process_full_box` to the superclass for a specific usage just for one subclass.
I looked into the `full_box` postprocessing more and figured we could update the boxes right after integrating with `Results` class.
Hence the function could be something like this:
```python
results = super().postprocess(preds, img, orig_imgs)
for result in results:
    full_box = torch.tensor([0, 0, result.orig_shape[1], result.orig_shape[0]], device=preds[0].device, dtype=torch.float32)
    idx = bbox_iou(full_box, result.boxes.xyxy, iou_thres=0.9, image_shape=result.orig_shape)
    if idx.numel() != 0:
        result.boxes.xyxy[idx] = full_box
return results
```

**Comment by Laughing-q** [comment]
@maycuatroi @glenn-jocher ok I made the updates I mentioned above. And also eliminated the `bbox_iou` function specifically added for fastsam module, currently it's using our universal `box_iou` to calculate iou values.

**Review by Laughing-q — APPROVED** [review]

**Comment by Laughing-q** [comment]
@glenn-jocher Please take a look at the `utils.md`, am I removing the reference function correctly? I'm not certain about this as you updated the docs page recently. Thanks!

**Comment by glenn-jocher** [comment]
@Laughing-q ah yes, it looks like you updated the Docs correctly!

For PRs on branches with permissions the docs should update automatically, for user PRs like this one the Docs fix CI will error, it's something I need to work on, but as a workaround I've been creating branches myself and merging user PRs into them and letting the Docs action run if it needs to, but we don't need to here as it's passing :)
